### PR TITLE
Feat/8703 add package dsg resource data

### PR DESCRIPTION
**Pull request by morhag90**
Close: https://github.com/amplication/private-issues/issues/226)

## PR Details

Add packages data to dsgResourceData and pass it to runPackageGenerator function in the build-manager-service
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: libs/util/code-gen-types/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@amplication/code-gen-types",
-  "version": "2.0.35",
+  "version": "2.0.36-beta.0",
```
Comment (yuval-hazaz) [inline libs/util/code-gen-types/package.json:3]: is that correct?

File: libs/util/code-gen-types/src/code-gen-types.ts [hunk 2]
```diff
@@ -225,6 +225,16 @@ export type PropertyTypeDef = Omit<models.PropertyTypeDef, "type"> & {
...
 
+export type Package = BlockOmittedFields<models.Package>;
+
+export type PackageFile = {
+  package: Package;
+  path: string;
+  originalFileChecksum: string;
+  fullContent: string;
+  diff: string;
+};
```
Comment (yuval-hazaz) [inline libs/util/code-gen-types/src/code-gen-types.ts:236]: why do we need this? (asking for knowledge)

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.ts [hunk 3]
```diff
@@ -21,10 +22,9 @@ export class BuildRunnerController {
   async onCodeGenerationSuccess(
     @Payload() dto: CodeGenerationSuccessDto
   ): Promise<void> {
-    await this.buildRunnerService.processBuildResult(
-      dto.resourceId,
-      dto.buildId, // jobBuildId
-      EnumJobStatus.Success
+    await this.buildRunnerService.runPackageGenerator(
+      dto.buildId,
+      dto.resourceId
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.ts:27]: let's talk about it

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: libs/util/code-gen-types/src/code-gen-types.ts (hunk 2 above)
Comment (morhag90) [reply libs/util/code-gen-types/src/code-gen-types.ts:236]: @yuval-hazaz For reusing this type in the future (maybe)

File: libs/util/code-gen-types/package.json (hunk 1 above)
Comment (morhag90) [reply libs/util/code-gen-types/package.json]: @yuval-hazaz bump version to : 2.0.36 (I will publish it after your final approve).

File: packages/amplication-build-manager/src/build-runner/build-runner.service.ts [hunk 4]
```diff
@@ -179,15 +215,7 @@ export class BuildRunnerService {
...
-        const successEvent: CodeGenerationSuccess.KafkaEvent = {
-          key: null,
-          value: { buildId, codeGeneratorVersion },
-        };
-
-        await this.producerService.emitMessage(
-          KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC,
-          successEvent
-        );
+        await this.runPackageGenerator(buildId, resourceId);
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.service.ts:218]: this should be conditional - only if there are packages

File: packages/amplication-build-manager/src/build-runner/build-runner.service.ts [hunk 5]
```diff
@@ -31,6 +32,41 @@ export class BuildRunnerService {
...
+    const successEvent: CodeGenerationSuccess.KafkaEvent = {
+      key: null,
+      value: { buildId, codeGeneratorVersion },
+    };
+
+    await this.producerService.emitMessage(
+      KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC,
+      successEvent
+    );
+  }
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.service.ts:47]: move this to a seperate function, and call it from here and from dsgCompleted when there are not packages

File: packages/amplication-build-manager/src/build-runner/build-runner.service.ts [hunk 6]
```diff
@@ -31,6 +32,40 @@ export class BuildRunnerService {
...
+      key: null,
+      value: { resourceId: resourceId, buildId: buildId, dsgResourceData },
+    };
+    await this.producerService.emitMessage(
+      KAFKA_TOPICS.PACKAGE_MANAGER_CREATE_REQUEST,
+      requestPackagesEvent
+    );
+  }
+
+  async dsgCompleted(buildId: string) {
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.service.ts:55]:
```suggestion
  async codeGenerationSuccess(buildId: string) {
```

File: packages/amplication-build-manager/src/build-runner/build-runner.service.ts [hunk 7]
```diff
@@ -31,6 +32,40 @@ export class BuildRunnerService {
...
 
+  async onPackageManagerCreateResponse(buildId: string) {
+    await this.dsgCompleted(buildId);
+  }
+
+  async runPackageGenerator(jobBuildId: string, resourceId: string) {
+    const buildId = this.buildJobsHandlerService.extractBuildId(jobBuildId);
+
+    const dsgResourceData =
+      await this.buildJobsHandlerService.extractDsgResourceData(jobBuildId);
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.service.ts:43]: you are calling extractDsgResourceData twice - maybe you can send the packages from the parent

File: packages/amplication-build-manager/src/build-runner/build-runner.service.ts [hunk 8]
```diff
@@ -31,6 +32,40 @@ export class BuildRunnerService {
     private readonly logger: AmplicationLogger
   ) {}
 
+  async onPackageManagerCreateResponse(buildId: string) {
+    await this.dsgCompleted(buildId);
+  }
+
+  async runPackageGenerator(jobBuildId: string, resourceId: string) {
```
Comment (yuval-hazaz) [inline packages/amplication-build-manager/src/build-runner/build-runner.service.ts:39]:
```suggestion
  async generatePackages(jobBuildId: string, resourceId: string) {
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**